src/components/CancelButton/CancelButton.js

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+
+import { Modal } from 'carbon-components-react';
+import StopFilled16 from '@carbon/icons-react/lib/stop--filled/16';
+
+import './CancelButton.scss';
+
+class CancelButton extends Component {
+  state = {
+    showDialog: false
+  };
+
+  handleCancel = () => {
+    const { name, onCancel } = this.props;
+    onCancel(name);
+    this.handleClose();
+  };
+
+  handleClose = () => {
+    this.setState({ showDialog: false });
+  };
+
+  handleShow = () => {
+    this.setState({ showDialog: true });
+  };
+
+  render() {
+    const closeIcon = <StopFilled16 />;
+    const { showDialog } = this.state;
+    const { type, name } = this.props;
+    return (
+      <>
```

Comment:
Carbon provides a ModalWrapper component which wraps a button, modal, and associated logic
http://react.carbondesignsystem.com/?path=/story/modalwrapper--transactional-passive-modal

Reply:
ignore this, ModalWrapper doesn't work well with icon triggers